### DISCUSSION TITLE
The CrossProbing sample application does not compile.

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#1998] [Q&A]
The CrossProbing sample located a:

`https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=cross-probing-sample&imodel=Bay+Town+Process+Plant`

...does not compile locally because is imports a non-existent npm package: @itwinjs-sandbox:

`import { AuthorizationClient, default3DSandboxUi, SampleIModels, useSampleWidget, ViewSetup } from "@itwinjs-sandbox";`

Are these intended to be usable samples?

**Comment by aruniverse** [comment]
Hi John,

The `@itwinjs-sandbox` is an internal package used by the showcase to abstract away generic logic not really relevant for the core apis that the sample is hoping to demonstrate.

I would take a look at this [tutorial](https://developer.bentley.com/tutorials/adding-showcase-widgets-to-your-itwin-viewer) which will help you go from a sample in the showcase to locally. 

Hopefully one of the showcase owners can help provide you more information.

@Josh-Schifter @StefanRetief @dustinlebsock

**Comment by StefanRetief** [comment]
Hi John,

Arun is correct. ```@itwinjs-sandbox``` is internal to the showcase.

You could copy out the widgets and supporting files out from the sample and use them within your project. In this case, you'll want to copy over the ```CrossProbingApi.ts``` file and the ```CrossProbingFrontstageProvider.tsx``` file. As for the contents of the ```CrossProbingApp.tsx```, the callbacks you'll want to copy over to your viewer file include the ```_oniModelReady``` and ```getFirst2DView```.

Lastly, you'll probably need a method to get a 3D ```ViewState``` given that the showcase supplies a ```ViewState``` for you, using ```ViewSetup.getDefaultView```. In this case, I have crafted this method:

```ts
const getFirst3DView = async (imodel: IModelConnection): Promise<ViewState> => {
    const defaultViewId = await imodel.views.queryDefaultViewId();
    return imodel.views.load(defaultViewId);
  };
```
